### ISSUE TITLE
Consider use of github organization for all Emacs typescript-repos

### ISSUE THREAD

**Issue by josteink** [ananthakumaran/tide#291]
Quickly summed up:

To encourage more contributors, we should make them feel welcome and empowered.

IME a good way to do this is providing push access early, and encourage them to push official branches, and while still "fresh" ask for review of these before letting them merge them to master themselves.

I'd argue that at least me and @lddubeau are active enough maintainers/developers for it to make sense for us to make such a call to provide push-access (and in the future, hopefully others too!)

But right now only @ananthakumaran has the power to do, but these days he seems less and less invested in both `tide` and `typescript-mode` .

IME open source projects often have changing leadership/ownership over time and it's therefore important to ensure a good base for the project to keep on growing.

A simple way to solve this is creating a GitHub organization, and migrating our repos there. This way more than one person can be a admin/owner of the repo and over time, ownership can become more organic/fluid.

Anyone explicitly positive to such a change? Anyone explicitly negative?

As a point of discussion, if we decide on making such an organization, what should the scope be?

* Should it only be typescript-related (`emacs-typescript`, `typescript-emacs`)?
* Should we be open to other programming modes too (`emacs-prog-modes`)?
* Other suggestions?

Opinions? Anyone?

CC: @ananthakumaran @lddubeau @MarcusRiemer @jack-williams @Ailrun @Ptival @Wilfred

**Comment by MarcusRiemer** [comment]
Just so this has been mentioned: I don't consider the current state to be "broken" or explicitly "unwelcoming" in any kind of way. But I do agree that there is room for improvement. I consider myself as a more or less passive bystander, but from my POV this does make sense. So this gets a :+1:  from me.

Or in a more poetic way: What seems to have started as a personal project from @ananthakumaran has left its fledgling years and is now ready to spread its wings and be taken over by a greater community. Thanks a lot for what you started @ananthakumaran, tide accompanied me through my masters degree that I started almost three years ago and is currently a reliable companion for my day to day work.

**Comment by lddubeau** [comment]
I think the relatively pressing issue here is to move the projects to an organization. It seems to me the related questions of scope, etc. can be dealt with later.

I'm all in favor of moving the projects to an organization. 

The one potential related issue that comes to my mind is this. Is there any administrative action related to how we publish packages that would also be useful to do? The package release framework we use is not something I've paid deep attention to. I just know if this were an `npm`-based project, we'd want to make sure the ownership of the `npm` packages is set so that we are not dependent on a single person.

**Comment by josteink** [comment]
> Is there any administrative action related to how we publish packages that would also be useful to do?

Package-release is done automatically by MELPA. Basically the MELPA recipie contains the Github-repo to monitor and MELPA does the rest automatically:

* new "unstable" builds on daily basis if there are new commits.
* new "stable" builds when version-tags are pushed to the repo.

Basically it's as close to zero-admin as you can get. Anyone with push-access to the repo will also have release-access.

**Comment by ananthakumaran** [comment]
Let's first decouple `tide` and `typescript-mode`

I am not the owner of the `typescript-mode` and the only reason it's
under my profile is due to the dependency between both of the
packages. Melpa maintainer wouldn't let me add typescript as a
dependency of tide without it also being present somewhere (melpa or
elpa). So I had to cleanup the repo and add it in melpa. I don't know
the codebase well and I am not much familiar with the syntax
parse/highlight apis as well. I am ok to transfer the ownership of typescript-mode to an
org.

Tide on the other hand is different matter. I started the project from
scratch and I still consider myself as the main [contributor](https://github.com/ananthakumaran/tide/graphs/contributors). I believe
I spend enough time to keep the project going. I am also open to give
commit access to more contributors.

To be frank about the project ownership, (not only about tide, but
most of the projects I have released under open source license), I
don't just share the code because I am altruistic, there are other
reasons like

1) signals the job market about my ability
2) shows the design holes when people try to use it for their usecases
3) get to collaborate with other people
4) full control to make any design decision

There are many others, but the point being, having the project under
my profile vs under different org is not same for me. I have more
incentives to keep it under my profile. This is not the first time
somebody has asked me to publish under different org.

There might be a stage where I might completely loose interest or the
project may outgrow me and others might disagree with me. In that case
community members can fork the repo and move on without me.

I don't think I have lost my interest in tide yet. Just to repeat
myself, I am more than willing to give access to people who wants to
contribute.

> To encourage more contributors, we should make them feel welcome and
> empowered.

I am trying to understand what's not being done properly now. To put
some stats, there are less than 50 commits made to the master branch
this year. I don't know if a project at this stage warrants seperate
org and other things.

**Comment by josteink** [comment]
> I am not the owner of the typescript-mode ... 

When you mention it, I do indeed remember. And this is also where I've seen your presence somewhat lacking, spurring this thought.

After filing this issue (on `tide`) I realized that *some* of the reasons were more applicable to `typescript-mode`, but decided not to delete and refile the issue.

Hopefully this hasn't caused too much confusion.

> I am ok to transfer the ownership of typescript-mode to an org.

Great!

I'll probably offload [ts-comint](https://github.com/josteink/ts-comint) too, so there's more than just `typescript-mode` there.

> Tide on the other hand is different matter. I started the project from scratch and I still consider myself as the main contributor. 

Indeed. Sorry if my comment in any way came of reducing your efforts here.

>  I don't just share the code because I am altruistic, there are other reasons 

I can definitely sympathize. I do the same for other projects, and I too have my reasons. Many of them overlapping yours.

>  the point being, having the project (tide) under my profile vs under different org is not same for me. I have more incentives to keep it under my profile. This is not the first time somebody has asked me to publish under different org.

Fair enough, and I honestly didn't know that.

It was mostly a suggestion for what I (on my end) would consider an improvement in the ability to self-organize. And I'm perfectly OK with people not always approving of my suggestions.

If you don't like it, you don't like it and that's OK 😄 

> I don't think I have lost my interest in tide yet. 

Noted. Again, sorry of this came off wrong, due to being filed under incorrect Github project.

> Just to repeat myself, I am more than willing to give access to people who wants to contribute. ...
> I am trying to understand what's not being done properly now.

Just to be clear: I don't think anything is *improper* now. But I considered an organization a way to improve contributors ability to self-organize.

But if you as the main developer/contributor still want to "own" `tide`, I can see how that's counter to your interests, and I respect that.

Speaking as a co-contributor/maintainer there are things I may like to do (either to `tide` or `typescript-mode`):

* Add CI builds (I did that, but needed the repo-owner to approve this in travis)
* Add security-scans (There are SASS for this, which may require repo-owner's approval)
* Add contributors more non-nonchalantly (especially in `typescript mode`), where they are needed the most, without having to bother you about it.

Other things which "the community" in theory might like to do:

* Create github pages for the repos and packages to document them better
* Etc.

Most of these changes requires full repo-owner access. And this kind of access can only be delegated to other people for repos belonging to organizations.

I see your point about `tide` from your personal employment marketing POV, and I'm 100% OK with that, but when I originally made this suggestion, it just made sense to me to keep transfer both `typescript-mode` and `tide` in one go.

Again: If that's not what you want to do, that's completely fine with me.

And finally: I hope I haven't offended anyone by suggestion these changes. They were all meant to further the community which has formed around these projects and repos. So hopefully no hard feelings 😄

**Comment by lddubeau** [comment]
@ananthakumaran, I had started writing a response but @josteink's own [response](https://github.com/ananthakumaran/tide/issues/291#issuecomment-449001443) overlaps with what I was going to write. So I'll say that I agree with what he wrote, and save myself a lot of writing.

I'll add that *on principle* I prefer projects with multiple active contributors to be held by organizations rather than individuals. So if someone proposes to move a project with multiple active contributors to an organization, I don't need any deep argumentation to agree. When @josteink asked, I did not look very closely at the reasons. I prefer organizations on principle because I've had to deal a number of times with projects where the owner became disinterested and that caused a lot of headaches. If done properly, an organization can lessen the problems. It's like buying insurance to protect against a future potential problem. (It does not mean the problem is happening *now*.)

This being said, I'm not militant about it.
